Segfault

Hi,

I just gave a try to this provider and encountered a segfault on the first try:

```
Stack trace from the terraform-provider-sshtunnel_v0.1.1 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x623b80]

goroutine 52 [running]:
github.com/johanneswuerbach/terraform-provider-sshtunnel/internal/provider.handleConnection({0x8f5d58, 0x4000358270}, 0x0?, {0x8faab8, 0x40003b0010}, {0x40001d6bf0?, 0x0?})
        github.com/johanneswuerbach/terraform-provider-sshtunnel/internal/provider/connection_ephemeral_resource.go:290 +0x150
created by github.com/johanneswuerbach/terraform-provider-sshtunnel/internal/provider.(*ConnectionEphemeralResource).createPortForward.func1 in goroutine 65
        github.com/johanneswuerbach/terraform-provider-sshtunnel/internal/provider/connection_ephemeral_resource.go:262 +0x80

Error: The terraform-provider-sshtunnel_v0.1.1 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

--- a/internal/provider/connection_ephemeral_resource.go
+++ b/internal/provider/connection_ephemeral_resource.go
@@ -286,6 +286,7 @@ func handleConnection(ctx context.Context, sshConn *ssh.Client, localConn net.Co
 	remoteConn, err := sshConn.Dial("tcp", remoteAddr)
 	if err != nil {
 		tflog.Error(ctx, "failed to dial remote connection", map[string]interface{}{"err": err})
+		return
 	}
 	defer remoteConn.Close()
 

